modules/stochastic_tools/doc/content/modules/stochastic_tools/install_pytorch.md

```diff
@@ -0,0 +1,146 @@
+# Install MOOSE with Pytorch C++ API (libtorch)
+
+The way one enables pytorch capabilities in MOOSE depends on
+the operating system (Linux or Mac) and if we use HPC or just a local workstation.
+
+!alert! note
+Before we review the main approaches, it is important to emphasize that
+linking MOOSE with libtorch on Linux machines is not support if the compiler stack has been built
+using a `libc` version below 2.27 (for `libtorch v 1.8+`)
+or 2.23 (for `libtorch v1.4-1.8`). Furthermore, we do not support `libtorch` versions below
+v1.4. To check your currently used libc version use the following command:
+
+```bash
+ldd --version
+```
+
+!alert-end!
+
+## Installation on a local Mac workstation
+
+The first step of this process is to set up the a suitable conda environment. For this follow
+the instructions [here](https://mooseframework.inl.gov/getting_started/installation/conda.html).
+Next, activate the conda environment (assume `moose-torch`), clone the repository
+ and navigate to the base folder of the
+stochastic tools module (let's say it is in `~/projects`):
+
+```bash
+conda activate moose-torch
+git clone https://github.com/grmnptr/moose.git moose-torch
+cd moose-torch
+git checkout torch-compile-19571
+cd ~/projects/moose/modules/stochastic_tools
+```
+
+As a next step, we download a precompiled version of `libtorch` which is suitable
+for our system. This is carried out by a bash script as:
+
+```bash
+./scripts/setup_libtorch.sh
+```
+
+!alert! note
+The desired version of libtorch can be set by the following
+argument:
+
+```bash
+./scripts/setup_libtorch.sh --version=1.8
+```
+
+!alert-end!
+
+which creates a `libtorch` folder in the stochastic tools root directory. This folder
+will contain the necessary header files and shared object files which can be used
+for dynamic linking.
+
+The last step is to compile our MOOSE with libtorch. For this, a non-unity build
+is used (which is considerably slower than the default unity build) in the
+following manner:
+
+```bash
+MOOSE_UNITY=false make -j 8
+```
+
+## Installation on a local Linux workstation
+
+The difference compared to Mac machines is that the moose conda packages (compiler stack) have been
+built using `libc` v2.12 meaning that it cannot be used to set up the framework with
+libtorch. For this reason, we need to install PETSc and Libmesh manually, assuming that
+the default compiler stack on the system uses libc 2.27+ (most modern distributions,
+such as Ubuntu actually do have suitable compilers). For this first, clone the
+repository, then use the corresponding scripts as:
+
+```bash
+git clone https://github.com/grmnptr/moose.git moose-torch
+cd moose-torch
+git checkout torch-compile-19571
+./scripts/update_and_rebuild_petsc.sh
+./scripts/update_and_rebuild_libmesh.sh
```

Comment:
You might point to our PETSc and libmesh install instructions  

https://mooseframework.inl.gov/getting_started/installation/install_moose.html

Reply:
Addressed!